src/main/java/com/iota/iri/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java

```diff
@@ -94,6 +96,17 @@
      */
     private TransactionRequester transactionRequester;
 
+    /**
+     * The actual start milestone index from which the node started from
+     */
+    private int milestoneStartIndex;
+
+    /**
+     * Used to keep track of the times when the last N milestones got applied:
+     * used to calculate the average time needed to apply a milestone.
+     */
+    private ArrayBlockingQueue<Long> lastMilestoneApplyTimes = new ArrayBlockingQueue<>(20);
```

Comment:
This also should be cleared after the node is synced...

What I think will be the best design for the new two fields is to create a private inner class `SyncProgressBar` (or whatever other appropriate name) that will hold those two fields.
Every time the node gets synced you clear them both.
All the methods that manipulate this progress bar will be in this class.

Of course `AsciiProgressBar` will remain a public utility class

Reply:
I am now hiding the variables in an inner class. States get reset on each application where the node is synced.